master/buildbot/test/unit/test_steps_source_svn.py

```diff
@@ -2092,7 +2092,10 @@ def _makeSUCTest(input, exp):
     test_dont_remove_port_other = _makeSUCTest(
         "https://foo.com:2093/bar", "https://foo.com:2093/bar")
     test_quote_funny_chars = _makeSUCTest(
-        "http://foo.com/\x10\xe6%", "http://foo.com/%10%E6%25")
+        "http://foo.com/\x10\xe6%",
+        ("http://foo.com/%10%E6%25",    # Python 2
+         "http://foo.com/%10%C3%A6%25"  # Python 3
```

Comment:
wait a minute.. I think this just means the svnUriCanonicalize does not work as expected with python3.

if for svn, url canonyfication of http://foo.com/\x10\xe6% is supposed to return http://foo.com/%10%E6%25, we cannot make it return http://foo.com/%10%C3%A6%25.

I think what can be assumed here is that the input of svnUriCanonicalize is supposed to be bytes (and not native string). Wouldn't it simplify the problem you had?